controllers/operator_controller.go

```diff
@@ -162,14 +162,35 @@ func (r *OperatorReconciler) reconcile(ctx context.Context, op *operatorsv1alpha
 		return ctrl.Result{}, err
 	}
 
-	// update operator status
+	// convert existing unstructured object into bundleDeployment for easier mapping of status.
+	existingTypedBundleDeployment := &rukpakv1alpha1.BundleDeployment{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(dep.UnstructuredContent(), existingTypedBundleDeployment); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// update operator status:
+	// 1. If the Operator "Ready" status is "Unknown": The status of successful bundleDeployment is unknown, requeue the request.
+	// 2. If the Operator "Ready" status is "True": Update the "successful resolution" status and return the result.
+	// 3. If the Operator "Ready" status is "False": There is error observed from Rukpak. Update the status accordingly.
+	status, message := verifyBDStatus(existingTypedBundleDeployment)
+	var reason string
+	if status == metav1.ConditionTrue || status == metav1.ConditionUnknown {
+		reason = operatorsv1alpha1.ReasonResolutionSucceeded
+	} else {
+		reason = operatorsv1alpha1.ReasonBundleDeploymentFailed
+	}
+
```

Comment:
Nit: Put this in it's own function, and have that function return the entire `metav1.Condition` object?

Then the `apimeta.SetStatusCondition` below looks a bit simpler:
```go
readyCondition := mapBDStatusToReadyCondition(existingBD)
apimeta.SetStatusCondition(&op.Status.Conditions, readyCondition)
```